libraries/utils/utils_global.h

```diff
@@ -49,7 +49,7 @@
 #if defined(STATICLIB)
 #  define UTILSSHARED_EXPORT
 #elif defined(UTILS_LIBRARY)
-#  define UTILSSHARED_EXPORT Q_DECL_EXPORT    /**< Q_DECL_EXPORT must be added to the declarations of symbols used when compiling a shared library. */
+#  define UTILSSHARED_EXPORT Q_DECL_EXPORT  __attribute__ ((visibility ("default")))  /**< Q_DECL_EXPORT must be added to the declarations of symbols used when compiling a shared library. */
```

Comment:
I do not understand what is happening here. Can you explain pls?

Reply:
visibility while linking libraries... what a wonderful subject to read at 1am...
https://gcc.gnu.org/wiki/Visibility

Reply:
I would like to put a hold on this change until I familiarize myself with what it exactly does. Also, if we decidde to add this, we should try changing it project wide. Maybe this is something for a future PR?

Reply:
I think it just optimizes the content of a dll so that only the used functions are prepared. And this way the load times are much shorter.